### PR TITLE
11-getArticlesQueryRefactor

### PR DESCRIPTION
**Pull request by lewispricey**
Adds order, sort-by and topics query's to the articles endpoint.

Not 100% on if it is safe to use $s for the column name & order but tried both I & L and neither seemed to work in that part of the query. Any advice would be appreciated :)

### CODE REVIEW

**Comment by lewispricey** [comment]
sorry, this looks to have pulled though last nights commit strangely as the comments are not still there locally.

Will re-save & push again now

File: models/articles.js [hunk 1]
```diff
@@ -28,14 +28,17 @@ exports.updateArticle = ({article_id}, body) => {
...
     return db
-    .query(`
-        SELECT articles.article_id, title, topic, articles.author, articles.created_at, articles.votes, COUNT(comment_id) AS comment_count 
-        FROM articles 
-        LEFT OUTER JOIN comments ON articles.article_id = comments.article_id 
-        GROUP BY articles.article_id 
-        ORDER BY articles.created_at DESC;`)
+    .query(queryString)
     .then(({rows}) => rows)
 }
```
Comment (corkesw) [inline models/articles.js:45]: It's worth adding some logic that checks that the sort_by is one of the valid columns... created_at, votes etc.. if it's not we can reject and there's no need to query the db. Also adds a layer of protection from SQL injection

File: models/articles.js [hunk 2]
```diff
@@ -28,17 +28,32 @@ exports.updateArticle = ({article_id}, body) => {
...
+const checkUserRequest = (querys) => {
+    const acceptedSorts = [undefined, "title", "topic", "author", "created_at", "votes", "article_id"]
+    if(!acceptedSorts.includes(querys.sort_by)){
+        return Promise.reject({status: 400, msg: "Invalid Request"})
+    }
+    const acceptedOrders = [undefined, "ASC", "asc", "desc", "DESC"]
+    if(!acceptedOrders.includes(querys.order)){
+        return Promise.reject({status: 400, msg: "Invalid Request"})
+    }
+}
```
Comment (corkesw) [inline models/articles.js]: Might be tidier to move this util function to the util folder

**Comment by corkesw** [comment]
Good to merge

**Pull request merged**